core/src/transport/circuit_breaking.rs

```diff
@@ -0,0 +1,383 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use crate::{transport::TransportError, Multiaddr, Transport};
+
+use fnv::FnvHashMap;
+use futures::prelude::*;
+use futures_timer::Delay;
+use parking_lot::Mutex;
+use pin_project::pin_project;
+use std::{
+    cmp,
+    collections::hash_map::Entry,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+    time::Duration,
+};
+
+/// Timeout set when opening CircuitBreaker.
+const INITIAL_TIMEOUT: Duration = Duration::from_secs(2);
```

Comment:
2 seconds seems like a long time on first failure. What is the reasoning for this high value?

Reply:
The value was taken from the original PR (linked in the description). Would 1 second be more reasonable?

Reply:
Off the top of my head, I would suggest 200ms.

Reply:
Addressed in 25d5e86.